sdk/python/kfp/dsl/dsl_utils.py

```diff
@@ -97,6 +98,8 @@ def _output_artifact_path_placeholder(output_key: str) -> str:
 def _output_parameter_path_placeholder(output_key: str) -> str:
   return "{{{{$.outputs.parameters['{}'].output_file}}}}".format(output_key)
 
+def _intput_executor_placeholder() -> str:
+  return "{{{{$}}}}"
```

Comment:
I wish we could avoid using this placeholder. The format is monolithic, non-granular and can be somewhat platform-dependent. There were also breaking changes in the past.
If possible, it might be beneficial to pass the relevant pieces of data explicitly.

Reply:
I see your point. Since this change is not a requirement for custom job component I'm moving it change out to a separate PR(#6057) to better discuss / plan for it.